python/whylogs/api/store/profile_store.py

```diff
@@ -1,51 +1,26 @@
 from abc import ABC, abstractmethod
-from datetime import datetime
-from typing import Any, Dict, List, Optional
+from datetime import timedelta
+from typing import Optional
 
-from whylogs.api.writer.writer import Writable
-from whylogs.core import DatasetProfile, Segment
-from whylogs.core.schema import DatasetSchema
-from whylogs.core.stubs import pd
+from whylogs.api.store.date_config import DateConfig
+from whylogs.core import DatasetProfileView
 
 
 class ProfileStore(ABC):
-    """
-    A holder object for profiles.
-    """
-
-    @abstractmethod
-    def get_matching_profile(
-        self,
-        obj: Any = None,
-        *,
-        pandas: Optional[pd.DataFrame] = None,
-        row: Optional[Dict[str, Any]] = None,
-        segment: Optional[Segment] = None,
-        schema: Optional[DatasetSchema] = None,
-    ) -> List[DatasetProfile]:
-        return [DatasetProfile(schema)]
-
     @abstractmethod
-    def read(
-        self,
-        id: Optional[str],
-        dataset_timestamp: Optional[datetime] = None,
-        segment: Optional[Segment] = None,
-        schema: Optional[DatasetSchema] = None,
-    ) -> Optional[DatasetProfile]:
-        return [DatasetProfile(schema)]
-
-    @abstractmethod
-    def write(
-        self,
-        data: Writable,
-        model_id: Optional[str] = None,
-        dataset_timestamp: Optional[datetime] = None,
-        segment: Optional[Segment] = None,
-        schema: Optional[DatasetSchema] = None,
-    ) -> None:
+    def get(self, date_config: DateConfig) -> DatasetProfileView:
         pass
 
     @abstractmethod
-    def query(self, *args: Any, **kwargs: Any) -> List[DatasetProfile]:
+    def write(self, profile: Optional[DatasetProfileView]) -> None:
         pass
```

Comment:
actually the `query` method was from Jamie's stubs 😄 
so if I have 2 profiles written today, using `base_name-{timestamp}-{random_string}`, how can I refer to it in order to get it? since we discussed that `id` would only be the `base_name` at the moment

Reply:
we can add query later, starting with this tighter scope looks reasonable.